Sources/PaymentsCore/Networking/APIClient.swift

```diff
@@ -4,22 +4,42 @@ import Foundation
 ///
 /// `APIClient` is the entry point for each payment method feature to perform API requests. It also offers convenience methods for API requests used across multiple payment methods / modules.
 public class APIClient {
+        
+    // MARK: - Internal Properties
 
     private var http: HTTP
     private let sessionID = UUID().uuidString.replacingOccurrences(of: "-", with: "")
     private let coreConfig: CoreConfig
+    
+    /// The `AnalyticsService` instance is static/shared so that only one sessionID is used.
+    /// The "singleton" has to be managed here because `AnalyticsService` has a dependency on `HTTP`.
+    ///
+    /// Exposed for testing.
+    weak var analyticsService: AnalyticsService? {
+        get { APIClient._analyticsService }
+        set { APIClient._analyticsService = newValue }
+    }
+    private static var _analyticsService: AnalyticsService?
```

Comment:
I think that makes sense for the spacing if we feel like it will cut down on confusion, we have a space on BT currently, but I could see how that could lead to confusion potentially. 

With the grouping, it was more around right now this file goes private > internal > private whereas I think our ideal would be public > internal > private. I think we could move `analyticsService` and `_analyticsService` above the other private vars and still keep these two grouped together to increase readability. Not a big deal either way we go, though!

Reply:
Ah gotcha - I pushed up a change, let me know what you think!